docs/components/empty-state.md

```diff
@@ -1,245 +1,323 @@
-</KEmptyState>
+<KEmptyState
+  action-button-text="Action"
+  message="Lorem ipsum dolor sit amet..."
+  title="Empty State Content"
+/>
 ```
 
 ## Props
 
-### ctaIsHidden
+### title
 
-Boolean value used to hide the call to action button.
+String to display as title.
 
-<KEmptyState cta-is-hidden>
-  <template v-slot:title>No content</template>
-  <template v-slot:message>You do not have any content here 😉️</template>
-</KEmptyState>
+<KEmptyState
+  title="Empty State Title"
+  action-button-text="Action"
+/>
 
 ```html
-<KEmptyState cta-is-hidden>
-  <template v-slot:title>No content</template>
-  <template v-slot:message>You do not have any content here 😉️</template>
-</KEmptyState>
+<KEmptyState
+  title="Empty State Title"
+  action-button-text="Action"
+/>
 ```
 
-You can also use this to move your call to action into the message text.
+### message
 
-<KEmptyState cta-is-hidden>
-  <template v-slot:title>No services</template>
-  <template v-slot:message><router-link to="/">Add a service</router-link> to begin proxying traffic.</template>
-</KEmptyState>
+Text under the title. Ideal for providing a detailed explanation to the user on why they are seeing this screen. Can also be [slotted](#default).
+
+<KEmptyState
+  message="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla nec purus feugiat, molestie ipsum et, consequat nibh."
+  title="Empty State Content"
+/>
 
 ```html
-<KEmptyState cta-is-hidden>
-  <template v-slot:title>No services</template>
-  <template v-slot:message><router-link>Add a service</router-link> to begin proxying traffic</template>
-</KEmptyState>
+<KEmptyState
+  message="Lorem ipsum dolor sit amet..."
+  title="Empty State Content"
+/>
 ```
 
-### ctaText
+### actionButtonText
 
-A string to be used as the text content of the call to action button.
+Button text under the message.
 
-<KEmptyState cta-text="Button text">
-  <template v-slot:title>No Content</template>
-  <template v-slot:message>You do not have any content here 😉️</template>
-</KEmptyState>
+:::tip NOTE
+The button **won't be rendered** if this prop is not provided.
+:::
+
+<KEmptyState
+  action-button-text="Refresh"
+  title="Empty State Action Button"
+  message="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla nec purus feugiat, molestie ipsum et, consequat nibh."
+/>
 
 ```html
-<KEmptyState cta-text="Button text">
-  <template v-slot:title>No Content</template>
-  <template v-slot:message>You do not have any content here 😉️</template>
-</KEmptyState>
+<KEmptyState
+  action-button-text="Refresh"
+  title="Empty State Action Button"
+  message="Lorem ipsum dolor sit amet..."
+/>
 ```
 
-### handleClick
+### actionButtonShowIcon
 
-A function that is passed as the click handler for the call to action button
+Boolean to control whether to show or hide add icon in action button. Helpful when using action button for prompting user to create a resource. Defaults to `false`.
```

Comment:
issue: the name of this prop doesn't correlate to what it's actually doing -- you're only adding/hiding the "plus" icon with the prop, but it's named as if it always hides/shows the icon.

Separately I still think this isn't ideal; we don't offer props for any other icons. Seems like over-optimizing for internal use-cases.

Reply:
Removed the new prop